enhancements/agent-installer/automated-workflow.md

```diff
@@ -0,0 +1,491 @@
+   separate manifest files with `NMState` content using input data and
+   templates.
+6. The orchestration tool runs `openshift-install create image`.
+7. The installer extracts a copy of the RHCOS ISO from the OpenShift
+   release payload.
+8. The installer combines the inputs provided by the user and produces
+   an Ignition config.
+9. The installer combines the Ignition config and the RHCOS ISO to
+   create the bootable image configured to run the assisted service or
+   agent based on the host that boots the image and writes the new
+   image and cluster credentials to disk.
+10. The orchestration tool copies the ISO to a location that the hosts
+    or VMs can boot from it. For bare metal, that will generally be a
+    web server visible to the management controllers in the bare metal
+    hosts (because the management controllers are unlikely to be on
+    the same network as the external interfaces for the cluster
+    hosts). For vSphere or other hypervisors, that may be a filesystem
+    on the hypervisor host.
+11. The orchestration tool configures the host to boot the ISO. For
+    bare metal, that means configuring the baseboard management
+    controller (BMC) in each host to expose the ISO through the
+    virtual media interface. For vSphere or other hypervisors, similar
+    settings of the VM need to be adjusted.
+12. The orchestration tool boots the host.
+13. The orchestration tool uses `openshift-install wait-for
+    install-complete` to watch the deployment progress and wait for it
+    to complete.
+14. On each host, the startup script in the image selects the correct
+    network settings based on the MAC addresses visible and applies
+    the network settings for each interface.
+15. On node0, the startup script in the image recognizes that it is on
+    node0 and launches the assisted service and the create-cluster
+    service to drive the deployment.
+16. On node0, the create-cluster service waits for the assisted
+    service to start, then creates an InfraEnv using the API.
+17. On node0, the create-cluster service waits for all of the known
+    agents to be registered with the service.
+18. On all nodes (including node0), the startup script in the image
+    connects to the assisted service on node0 to find the InfraEnv ID
+    for the cluster being deployed (there will only be 1). After it
+    has the ID, the startup script launches the assisted agent with
+    that ID and configured to connect to the assisted service on
+    node0.
+19. All of the agents register their hosts with the assisted service.
+20. The assisted service performs its standard validations for all of the
+    hosts.
+21. On node0, the create-cluster service sees that all known agents
+    have registered and the hosts have passed validation. It uses the
+    assisted service API to trigger the deployment to proceed.
+22. The assisted service runs the OpenShift installer and performs the
```

Comment:
I think this step actually talking about the agent running coreos-installer, so perhaps the wording can be clarified?

Reply:
I can update the wording, but I thought it ran the installer to get the manifests. Maybe it would be best to remove any reference to what it actually runs, since the point is that it just does what it does today.